### ISSUE TITLE
Feature: Avoid mutation of the option_object passed to the jslint function

### ISSUE THREAD

**Issue by nikoskalogridis** [douglascrockford/JSLint#202]
It seems that the jslint function mutates the object passed to it in its option_object parameter.

**Comment by douglascrockford** [comment]
Where does it mutate?

**Comment by nikoskalogridis** [comment]
```javascript
var options = {fudge: true};
jslint("/*jslint node*/", options);
console.log(options);
// => {fudge: true, node: true}
```

**Comment by gaborsar** [comment]
`option` set at:
https://github.com/douglascrockford/JSLint/blob/master/jslint.js#L4781

`option` mutated at:
https://github.com/douglascrockford/JSLint/blob/master/jslint.js#L866
https://github.com/douglascrockford/JSLint/blob/master/jslint.js#L872
https://github.com/douglascrockford/JSLint/blob/master/jslint.js#L884

**Comment by nikoskalogridis** [comment]
Also passing a frozen object as an options parameter would break execution of jslint

**Comment by douglascrockford** [comment]
Thanks. Please try it now.

**Comment by nikoskalogridis** [comment]
Thanks, so I guess es5.1 jacascript engines are not supported any more? Object.assign is an es2015 feature.

**Comment by douglascrockford** [comment]
It is a tool for programmers. I think programmers should be able to keep up. My interest is in moving us forward at a prudent rate.

**Comment by benquarmby** [comment]
I agree JSLint editor and platform integrators will keep up. It may be worth making an announcement since it will affect the choice of JavaScript engine.
